lib/core/src/Cardano/Wallet.hs

```diff
@@ -567,19 +573,17 @@ restoreBlocks ctx wid blocks nodeTip = do
         let k = bp ^. #getEpochStability
         let localTip = currentTip $ NE.last cps
 
-        meta' <- liftIO $ calculateMetadata bp localTip meta
         let unstable = sparseCheckpoints k (nodeTip ^. #blockHeight)
         forM_ (NE.init cps) $ \cp -> do
             let (Quantity h) = currentTip cp ^. #blockHeight
             when (fromIntegral h `elem` unstable) (makeCheckpoint cp)
         makeCheckpoint (NE.last cps)
         DB.prune db (PrimaryKey wid)
         DB.putTxHistory db (PrimaryKey wid) txs
-        DB.putWalletMeta db (PrimaryKey wid) meta'
 
         liftIO $ do
             logInfo tr $
-                pretty meta'
+                pretty meta
```

Comment:
This will now report the metadata of the checkpoint before we applied blocks :/

Reply:
Yes - the `meta` is not used in `restoreBlocks` aside from this `logInfo` which I in doubt opted to preserve.

Reply:
We should preserve it yes, but also add an extra one with the sync progress!